content/blog/run-deepseek-on-aws-ec2-using-pulumi/index.md

```diff
@@ -0,0 +1,502 @@
+
+```typescript
+{{< example-program-snippet path="deepseek-ollama" language="typescript" from="95" to="136" >}}
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+{{< example-program-snippet path="deepseek-ollama" language="python" from="106" to="142" >}}
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+{{< example-program-snippet path="deepseek-ollama" language="go" from="1" to="13" >}}
+                // omitted for brevity
+{{< example-program-snippet path="deepseek-ollama" language="go" from="141" to="204" >}}
+{{< example-program-snippet path="deepseek-ollama" language="go" from="205" to="208" >}}
+```
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+```csharp
+{{< example-program-snippet path="deepseek-ollama" language="csharp" from="1" to="12" >}}
+            // omitted for brevity
+{{< example-program-snippet path="deepseek-ollama" language="csharp" from="142" to="219" >}}
+{{< example-program-snippet path="deepseek-ollama" language="csharp" from="220" to="225" >}}
+```
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+```yaml
+{{< example-program-snippet path="deepseek-ollama" language="yaml" from="114" to="141" >}}
+```
+
+{{% /choosable %}}
+
+#### Install Ollama and Run DeepSeek
+
+After we set up all the infrastructure needed for our GPU-powered EC2 instance, we can install Ollama and run DeepSeek. This will all be done as part of the user data script we pass to the EC2 instance.
+
+In the `runcmd` section of the user data script, we will install the necessary packages, download the NVIDIA GRID drivers from S3, install Docker, and run the Ollama and Open WebUI containers.
+
+```yaml
+#cloud-config
+users:
+- default
+
+package_update: true
+
+packages:
+- apt-transport-https
+- ca-certificates
+- curl
+- openjdk-17-jre-headless
+- gcc
+
+runcmd:
+- yum install -y gcc kernel-devel-$(uname -r)
+- aws s3 cp --recursive s3://ec2-linux-nvidia-drivers/latest/ .
+- chmod +x NVIDIA-Linux-x86_64*.run
+- /bin/sh ./NVIDIA-Linux-x86_64*.run --tmpdir . --silent
+- touch /etc/modprobe.d/nvidia.conf
+- echo "options nvidia NVreg_EnableGpuFirmware=0" | sudo tee --append /etc/modprobe.d/nvidia.conf
+- yum install -y docker
+- usermod -a -G docker ec2-user
+- systemctl enable docker.service
+- systemctl start docker.service
+- curl -s -L https://nvidia.github.io/libnvidia-container/stable/rpm/nvidia-container-toolkit.repo | sudo tee /etc/yum.repos.d/nvidia-container-toolkit.repo
+- yum install -y nvidia-container-toolkit
+- nvidia-ctk runtime configure --runtime=docker
+- systemctl restart docker
+- docker run -d --gpus=all -v ollama:/root/.ollama -p 11434:11434 --name ollama --restart always ollama/ollama
+- sleep 120
+- docker exec -it ollama ollama pull deepseek-r1:7b
+- docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
+```
+
+{{< related-posts >}}
```

Comment:
Should the related posts be more AI focused? Its pulling some weird ones that don't have a relation with much in the article